components/style/viewport.rs

```diff
@@ -685,6 +685,7 @@ impl MaybeNew for ViewportConstraints {
         let initial_viewport = device.au_viewport_size();
 
         // TODO(emilio): Stop cloning `ComputedValues` around!
+        #[cfg(not(feature = "gecko"))]
         let context = Context {
```

Comment:
Changed to make `computed::Context` hold `&Device` unconditionally.  This simplified the code a lot; thanks.